url2bc31e3489_ramonli_RabbitMQResearch_tgz-pJ8-com_mpos_lottery_te_thirdpartyservice_amqp_protobuf_CancellationMessagePulisherJ8/src/com/mpos/lottery/te/thirdpartyservice/amqp/TeTransactionMessage.java

```diff
@@ -0,0 +1,6710 @@
+      }
+
+      public com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale getDefaultInstanceForType() {
+        return com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale.getDefaultInstance();
+      }
+
+      public com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale build() {
+        com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale result = buildPartial();
+        if (!result.isInitialized()) {
+          throw newUninitializedMessageException(result);
+        }
+        return result;
+      }
+
+      public com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale buildPartial() {
+        com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale result = new com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale(this);
+        int from_bitField0_ = bitField0_;
+        int to_bitField0_ = 0;
+        if (((from_bitField0_ & 0x00000001) == 0x00000001)) {
+          to_bitField0_ |= 0x00000001;
+        }
+        if (transactionBuilder_ == null) {
+          result.transaction_ = transaction_;
+        } else {
+          result.transaction_ = transactionBuilder_.build();
+        }
+        if (((from_bitField0_ & 0x00000002) == 0x00000002)) {
+          to_bitField0_ |= 0x00000002;
+        }
+        result.multiDraw_ = multiDraw_;
+        if (((from_bitField0_ & 0x00000004) == 0x00000004)) {
+          to_bitField0_ |= 0x00000004;
+        }
+        result.totalAmount_ = totalAmount_;
+        if (((from_bitField0_ & 0x00000008) == 0x00000008)) {
+          to_bitField0_ |= 0x00000008;
+        }
+        result.serialNo_ = serialNo_;
+        if (((from_bitField0_ & 0x00000010) == 0x00000010)) {
+          to_bitField0_ |= 0x00000010;
+        }
+        if (gameInstanceBuilder_ == null) {
+          result.gameInstance_ = gameInstance_;
+        } else {
+          result.gameInstance_ = gameInstanceBuilder_.build();
+        }
+        if (entriesBuilder_ == null) {
+          if (((bitField0_ & 0x00000020) == 0x00000020)) {
+            entries_ = java.util.Collections.unmodifiableList(entries_);
+            bitField0_ = (bitField0_ & ~0x00000020);
+          }
+          result.entries_ = entries_;
+        } else {
+          result.entries_ = entriesBuilder_.build();
+        }
+        result.bitField0_ = to_bitField0_;
+        onBuilt();
+        return result;
+      }
+
+      public Builder mergeFrom(com.google.protobuf.Message other) {
+        if (other instanceof com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale) {
+          return mergeFrom((com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale)other);
+        } else {
+          super.mergeFrom(other);
+          return this;
+        }
+      }
+
+      public Builder mergeFrom(com.mpos.lottery.te.thirdpartyservice.amqp.TeTransactionMessage.Sale other) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.